How to call esp_ble_gap_set_rand_addr?

It's in [BLEAdvertising.cpp](https://github.com/espressif/arduino-esp32/blob/master/libraries/BLE/src/BLEAdvertising.cpp#L286)

```bash
  = note: [ldproxy] Running ldproxy
          Error: Linker /Users/lz1998/RustroverProjects/evil-apple-juice-esp32-rs/.embuild/espressif/tools/riscv32-esp-elf/esp-12.2.0_20230208/riscv32-esp-elf/bin/riscv32-esp-elf-gcc failed: exit status: 1
          STDERR OUTPUT:
          /Users/lz1998/RustroverProjects/evil-apple-juice-esp32-rs/.embuild/espressif/tools/riscv32-esp-elf/esp-12.2.0_20230208/riscv32-esp-elf/bin/../lib/gcc/riscv32-esp-elf/12.2.0/../../../../riscv32-esp-elf/bin/ld: /Users/lz1998/RustroverProjects/evil-apple-juice-esp32-rs/target/riscv32imc-esp-espidf/debug/deps/libesp32_nimble-8f23e8b04dddce3f.rlib(esp32_nimble-8f23e8b04dddce3f.esp32_nimble.671d7f6fcb67d447-cgu.05.rcgu.o): in function `esp32_nimble::ble_device::BLEDevice::_set_own_addr_type':
          /Users/lz1998/.cargo/git/checkouts/esp32-nimble-fc9d2c54a3bfa7fe/8250ca8/src/ble_device.rs:(.text._ZN12esp32_nimble10ble_device9BLEDevice18_set_own_addr_type17h1ce402db9c242dddE+0x3c): undefined reference to `ble_hs_pvcy_rpa_config'
          collect2: error: ld returned 1 exit status
          
          
          Stack backtrace:
             0: std::backtrace_rs::backtrace::libunwind::trace
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5
             1: std::backtrace_rs::backtrace::trace_unsynchronized
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
             2: std::backtrace::Backtrace::create
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/backtrace.rs:331:13
             3: std::backtrace::Backtrace::capture
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/backtrace.rs:297:9
             4: anyhow::error::<impl anyhow::Error>::msg
             5: ldproxy::main
             6: std::sys_common::backtrace::__rust_begin_short_backtrace
             7: std::rt::lang_start::{{closure}}
             8: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/core/src/ops/function.rs:284:13
             9: std::panicking::try::do_call
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/panicking.rs:552:40
            10: std::panicking::try
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/panicking.rs:516:19
            11: std::panic::catch_unwind
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/panic.rs:142:14
            12: std::rt::lang_start_internal::{{closure}}
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/rt.rs:148:48
            13: std::panicking::try::do_call
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/panicking.rs:552:40
            14: std::panicking::try
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/panicking.rs:516:19
            15: std::panic::catch_unwind
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/panic.rs:142:14
            16: std::rt::lang_start_internal
                       at /rustc/dd430bc8c22f57992ec1457a87437d14283fdd65/library/std/src/rt.rs:148:20
            17: std::rt::lang_start
          
  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
```

--- a/src/ble_device.rs
+++ b/src/ble_device.rs
@@ -10,9 +10,12 @@ type BLEAdvertising = crate::BLEAdvertising;
 #[cfg(esp_idf_bt_nimble_ext_adv)]
 type BLEAdvertising = crate::BLEExtAdvertising;
 
-#[cfg(esp32)]
 extern "C" {
   fn ble_store_config_init();
+}
+
+#[cfg(esp32)]
+extern "C" {
   fn ble_hs_pvcy_rpa_config(enable: u8) -> core::ffi::c_int;
 }
 #[cfg(esp32)]

--- a/src/ble_device.rs
+++ b/src/ble_device.rs
@@ -10,12 +10,16 @@ type BLEAdvertising = crate::BLEAdvertising;
 #[cfg(esp_idf_bt_nimble_ext_adv)]
 type BLEAdvertising = crate::BLEExtAdvertising;
 
+#[cfg(esp32)]
 extern "C" {
   fn ble_store_config_init();
   fn ble_hs_pvcy_rpa_config(enable: u8) -> core::ffi::c_int;
 }
+#[cfg(esp32)]
 const NIMBLE_HOST_DISABLE_PRIVACY: u8 = 0x00;
+#[cfg(esp32)]
 const NIMBLE_HOST_ENABLE_RPA: u8 = 0x01;
+#[cfg(esp32)]
 const NIMBLE_HOST_ENABLE_NRPA: u8 = 0x02;
 
 static mut BLE_DEVICE: Lazy<BLEDevice> = Lazy::new(|| {
@@ -197,6 +201,7 @@ impl BLEDevice {
   }
 
   /// Set the own address type to non-resolvable random address.
+  #[cfg(esp32)]
   pub fn set_own_addr_type_to_non_resolvable_random(&mut self) {
     self._set_own_addr_type(OwnAddrType::Random, true);
   }
@@ -206,10 +211,12 @@ impl BLEDevice {
       OWN_ADDR_TYPE = own_addr_type;
       match own_addr_type {
         OwnAddrType::Public => {
+          #[cfg(esp32)]
           ble_hs_pvcy_rpa_config(NIMBLE_HOST_DISABLE_PRIVACY);
         }
         OwnAddrType::Random => {
           self.security().resolve_rpa();
+          #[cfg(esp32)]
           ble_hs_pvcy_rpa_config(if use_nrpa {
             NIMBLE_HOST_ENABLE_NRPA
           } else {
@@ -218,6 +225,7 @@ impl BLEDevice {
         }
         OwnAddrType::RpaPublicDefault | OwnAddrType::RpaRandomDefault => {
           self.security().resolve_rpa();
+          #[cfg(esp32)]
           ble_hs_pvcy_rpa_config(NIMBLE_HOST_ENABLE_RPA);
         }
       }

